lib/cartopy/io/img_tiles.py

```diff
@@ -30,58 +30,25 @@
 
 from __future__ import (absolute_import, division, print_function)
 
+from abc import ABCMeta, abstractmethod
+import warnings
+
 from PIL import Image
 import shapely.geometry as sgeom
 import numpy as np
 import six
-import warnings
 
 import cartopy.crs as ccrs
 
 
-class GoogleTiles(object):
+class GoogleWTS(six.with_metaclass(ABCMeta, object)):
```

Comment:
I love that this is getting the `WTMS` in the name but is it still worth keeping "Google" in there? Why not taking this opportunity to make the class name more generic? I believe `GoogleWTS` may throw users, who don't read the docs, off by thinking it only works for Google Tiles.

Reply:
A fine observation. I agree entirely with the sentiment, but it is really hard to find a name that doesn't include the word Google - this isn't a general WMTS implementation AFAICT - it really does build upon Google mercator and uses (as far as I can tell) a non-standard y indexing scheme.

I suspect there are further abstractions that would be useful (there is a 'Microsoft' Quadtree implementation in here that really only needs a small subset of the GoogleWTS class's behaviour).

In short - I'm 100% in favour of finding a name that represents the scheme that Google implemented (I think it was them at least), but at present don't have a better suggestion.